app/build.gradle

```diff
@@ -12,21 +16,50 @@ android {
         targetSdkVersion 28
         versionCode 1
         versionName "1.0"
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+
+    kapt {
+        generateStubs = true
+    }
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.android.support:appcompat-v7:28.0.0'
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    implementation 'androidx.appcompat:appcompat:1.1.0-alpha03'
+    implementation 'androidx.core:core-ktx:1.1.0-alpha05'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.google.android.material:material:1.0.0-rc01'
+
+    //KOIN
+    implementation "org.koin:koin-androidx-viewmodel:$koin_version"
+
+    //ROOM
+    implementation "androidx.room:room-runtime:$room_version"
```

Comment:
La librería de Room no iría en el módulo de data?

Osea yo me crearía tres modulos independientes (presentation, domain y data) y cada uno con su gradle y sus dependencias.
No se como lo verá el resto, a mi personalmente me gusta más hacerlo así que no todo en un modulo app y separado por paquetes.

Reply:
¿Room soporta multimódulos? No sé cómo funcionará Room porque nunca lo he usado, pero por ejemplo con ObjectBox, que sí lo he usado, por el momento no tiene soporte para multimódulos, aunque está propuesto, así que no sé si será por ese motivo que esté ahí incluido.

Reply:
He encontrado esto:

[https://stackoverflow.com/questions/49954279/had-anyone-success-with-using-room-over-multi-module-kotlin-setup](url)

Igual es por eso.

Reply:
Óscar, hay alguna razón en especial por la que te guste tener todo en módulos diferentes con sus gradles etc?.

El tema de llamar al paquete data o lo para mi no tiene importancia. Le puse IO de entrada salida y ahí subdividir sinhace falta para especializar. 

En cualquier caso, a ver qué opinan los demás, yo soy de la opinión de aplicar el principio KISS

Reply:
A mi me gusta separarlo por módulos porque a la hora de trabajar con ellos ves mucho mejor que dependencias existen entre los módulos (siguiendo el clean architecture) y al tener sus gradles lo veo más organizado que cada módulo use las librerías que debe usar.
A mí me ha ayudado a respetar más el clean architecture en los proyectos.

Reply:
@vcanadao Yo tambien lo veo como dice Oscar, es mas no veo el sentido de meter Room en esta PR, ya que va de Koin, a no ser que sea imprescindible para poder usar Koin(que no lo se).
Tampoco veo que necesitemos las coroutines, ni constrains, ni mockito...

Reply:
Cómo queráis,  si todos lo veis bien, mete mano con tu idea y sube los  cambios.  


A mi me gusta de principio tener todo en un módulo agrupados por paquetes. El nombre que quieras darle es sólo semántica, puedes tener app usercases dominios datos o los nombres que quiera mientras sigas la filosofía o metodología. 

Pero vamos, lo que queramos

Reply:
El tema es hacerlo con modulos diferentes, ademas de ver el tema de dependecias mas sencillo, la compilacion se ejecuta por modulos y los tiempos se reducen. Creo que es eso por lo que lo dice Oscar

Reply:
Si queréis lo hacemos así, 
aunque yo creo que en eta fase del proyecto es agregar complejidad innecesaria y la ganancia de performance a la hora de compilar es irrisoria, y más a estas alturas del proyecto. Pero ya digo si la mayoría lo ve así, por mi etá bien.

Reply:
Yo más que nada porque veo más sencillo crear los módulos ahora, dejarlos ya con sus gradles y a partir de ahi irles metiendo mano cuando lo necesitemos, que crear todo en uno y luego modularizarlo. Eso lo hice yo hace poco y es más dolor de huevos.